packages/plugin-api/src/utils/tree.ts

```diff
@@ -329,3 +329,87 @@ export const transformTree = <L, G>(
 
   return tree;
 };
+
+/**
+ * Creates tree by Title path
+ * @param tests
+ * @param getTitlePath
+ * @param getLeaf
+ */
+export const createTreeByTitlePath = <T extends Partial<TestResult>>(
+  tests: T[],
+  getTitlePath: (item: T) => string[],
+  getLeaf: (item: T) => any & { nodeId: string; status: string },
+): {
+  root: { groups: string[]; leaves: string[] };
+  groupsById: Record<string, any>;
+  leavesById: Record<string, any>;
+} => {
+  const root: { groups: string[]; leaves: string[] } = { groups: [], leaves: [] };
+  const groupsById: Record<string, any> = {};
+  const leavesById: Record<string, any> = {};
+  const groupPathToId = new Map<string, string>();
+  let groupOrder = 1;
+
+  for (const test of tests) {
+    const path = getTitlePath(test);
+    if (!Array.isArray(path) || path.length === 0) {
+      continue;
+    }
+
+    let parentGroupId: string | undefined;
+    const pathParts: string[] = [];
+
+    for (const part of path) {
+      pathParts.push(part);
+      const key = pathParts.join("/");
```

Comment:
Collisions are possible if there is `/` in a title path. For example, the following four title paths:

  - `["foo/bar/baz"]`
  - `["foo/bar", "baz"]`
  - `["foo", "bar/baz"]`
  - `["foo", "bar", "baz"]`

...will all give the `foo/bar/baz` key.

Let's use `JSON.stringify` instead. It's collision-free:

```js
const key = JSON.stringify(pathParts);
```